src/panels/LogPanel/ScriptLogPanel.tsx

```diff
@@ -0,0 +1,76 @@
+import { useCallback, useEffect, useState } from 'react';
+import { ActionIcon, Group, Tooltip } from '@mantine/core';
+
+import { useOpenSpaceApi } from '@/api/hooks';
+import { FilterList } from '@/components/FilterList/FilterList';
+import { wordBeginningSubString } from '@/components/FilterList/util';
+import { ReplayIcon } from '@/icons/icons';
+
+import { ScriptLogEntry } from './ScriptLogEntry';
+
+// TODO (anden88 2025-04-17): Remove when lists from OpenSpace don't have the
+// ['1']: value, ['2']: value format
+type FileLines = Record<string, string> | undefined;
+
+export function ScriptLogPanel() {
+  const luaApi = useOpenSpaceApi();
+  const [scriptLogEntries, setScriptLogEntries] = useState<string[]>([]);
+
+  const fetchScriptLogEntries = useCallback(async () => {
+    // eslint-disable-next-line no-template-curly-in-string
+    const fileName = await luaApi?.absPath('${LOGS}/ScriptLog.txt');
+
+    if (!fileName) {
+      return;
+    }
+    const data = (await luaApi?.readFileLines(fileName)) as FileLines;
+
+    if (!data) {
+      return;
+    }
+
+    setScriptLogEntries(
+      Object.values(data)
+        .filter((script) => script)
+        .reverse()
+    );
+  }, [luaApi]);
+
+  useEffect(() => {
+    fetchScriptLogEntries();
+  }, [fetchScriptLogEntries]);
```

Comment:
in other places where we fetch we have placed the function within the hook itself. might be nicer for consistency 🤷